### ISSUE TITLE
Feature Request: Display non-Civitai LoRAs' base model and training data from .safetensor metadata

### ISSUE THREAD

**Issue by lazybuttalented** [willmiao/ComfyUI-Lora-Manager#849]
It would be very useful if LoRA Manager could read and display the training metadata embedded in .safetensors LoRA files.

I have many LoRAs trained with tools like kohya-ss include useful metadata fields such as:

ss_base_model_version

ss_sd_model_name

ss_network_dim

ss_resolution

ss_num_train_images

ss_epoch

This information can be extracted directly from the safetensors metadata without loading the model weights.

I my memory in previous version of Lora Manager we had that function and the current version we cant get that.

**Comment by lazybuttalented** [comment]
<img width="626" height="919" alt="Image" src="https://github.com/user-attachments/assets/3ca0d9ad-487a-4f86-bcb5-3b7427b2f272" />
As you can see the Lora manager is listing it in Unknown Base Model. But I have the metadata below in it: 

`python -c "from safetensors import safe_open; f=safe_open('martha25.safetensors', framework='pt'); print(f.metadata())"
{'ss_fp8_base': 'False', 'ss_zero_terminal_snr': 'True', 'ss_lr_scheduler': 'cosine_with_restarts', 'ss_adaptive_noise_scale': 'None', 'ss_cache_latents': 'True', 'ss_mixed_precision': 'fp16', 'ss_loss_type': 'l2', 'ss_steps': '1596', 'ss_clip_skip': '1', 'ss_training_started_at': '1772849058.115291', 'ss_sd_scripts_commit_hash': '53fa3ca2f6205a5c2be8a5447c6f6251f10f6f5d', 'ss_num_reg_images': '0', 'ss_training_comment': 'None', 'ss_network_alpha': '32.0', 'modelspec.encoder_layer': '1', 'ss_multires_noise_iterations': '0', 'ss_network_args': '{"conv_dim": "32", "conv_alpha": "16", "factor": "8", "dropout": null}', 'ss_session_id': '3087931878', 'modelspec.title': 'lora', 'ss_lr_warmup_steps': '0.0', 'ss_caption_dropout_rate': '0.0', 'ss_prior_loss_weight': '1.0', 'ss_learning_rate': '2e-06', 'ss_huber_schedule': 'snr', 'ss_caption_tag_dropout_rate': '0.0', 'ss_max_train_steps': '1596', 'modelspec.implementation': 'https://github.com/Stability-AI/generative-models', 'ss_min_snr_gamma': '0.0', 'ss_debiased_estimation': 'False', 'ss_fp8_base_unet': 'False', 'ss_multires_noise_discount': '0.0', 'ss_unet_lr': '0.0001', 'ss_tag_frequency': '{"image_dir": {"martha25": 54, "1girl": 54, "solo": 53, "long hair": 53, "breasts": 47, "looking at viewer": 47, "black hair": 54, "navel": 28, "cleavage": 35, "brown eyes": 52, "medium breasts": 31, "sitting": 12, "underwear": 12, "panties": 9, "indoors": 23, "dark skin": 27, "bra": 11, "white panties": 4, "dark-skinned female": 24, "lips": 47, "bed": 11, "arm support": 4, "on bed": 8, "underwear only": 9, "lingerie": 10, "lace trim": 5, "lace": 7, "freckles": 7, "white bra": 5, "realistic": 54, "lace panties": 6, "smile": 24, "barefoot": 12, "teeth": 3, "midriff": 7, "pants": 7, "necklace": 2, "grin": 10, "feet": 6, "crop top": 6, "pillow": 1, "window": 4, "toes": 4, "sunlight": 2, "pajamas": 1, "nose": 3, "bedroom": 6, "black panties": 4, "black bra": 5, "standing": 12, "full body": 9, "carpet": 1, "swimsuit": 23, "bikini": 23, "outdoors": 12, "day": 14, "water": 10, "black bikini": 12, "highleg": 13, "thigh gap": 1, "front-tie top": 3, "pool": 6, "highleg bikini": 11, "poolside": 4, "long sleeves": 4, "shorts": 5, "sweater": 4, "sleeves past wrists": 1, "short shorts": 5, "white shorts": 4, "white sweater": 4, "kitchen": 2, "photo background": 8, "sink": 1, "skirt": 2, "shirt": 4, "jewelry": 6, "white shirt": 2, "collared shirt": 2, "black skirt": 2, "chair": 3, "pencil skirt": 2, "office lady": 2, "computer": 1, "office": 1, "dress": 3, "bare shoulders": 3, "sleeveless": 1, "black footwear": 1, "black dress": 1, "high heels": 2, "hand on hip": 3, "sleeveless dress": 2, "short dress": 1, "table": 4, "plant": 3, "building": 5, "couch": 4, "city": 6, "potted plant": 1, "cityscape": 4, "skyscraper": 1, "open clothes": 2, "abs": 1, "bikini top only": 1, "white pants": 1, "grey pants": 1, "green bikini": 1, "sky": 1, "white dress": 2, "sunset": 3, "railing": 1, "1boy": 1, "solo focus": 1, "cup": 2, "crossed legs": 2, "cafe": 1, "tree": 3, "beach": 3, "blue bikini": 2, "palm tree": 1, "ocean": 4, "white bikini": 3, "watercraft": 2, "ship": 2, "boat": 2, "warship": 2, "ass": 1, "thong": 1, "shadow": 4, "walking": 6, "phone": 1, "sandals": 2, "cellphone": 1, "road": 3, "street": 2, "sand": 1, "waves": 1, "hood": 1, "hoodie": 1, "black pants": 2, "hood down": 1, "black hoodie": 1, "cropped hoodie": 1, "jacket": 1, "black shirt": 1, "denim": 3, "ground vehicle": 1, "motor vehicle": 1, "jeans": 2, "car": 1, "denim jacket": 1, "bangs": 1, "closed mouth": 2, "stomach": 1, "bare legs": 1, "on couch": 1, "cropped sweater": 2, "tank top": 2, "denim shorts": 1, "white tank top": 1, "upper body": 9, "earrings": 5, "outstretched arms": 5, "cowboy shot": 1, "highleg swimsuit": 1, "simple background": 3, "white background": 3, "profile": 2, "collarbone": 2, "nude": 2, "portrait": 1, "black eyes": 1, "thighs": 1, "blurry": 1, "book": 2, "lamp": 2, "navel piercing": 2, "large breasts": 2, "arms at sides": 1, "armpits": 1, "arm up": 1, "sports bra": 1, "gym": 1, "kneeling": 1, "faucet": 1, "holding": 1, "wooden floor": 1, "open book": 1, "reading": 1, "one eye closed": 1, "spread legs": 1, "toned": 1, "tan": 2, "highleg panties": 1, "lace bra": 1, "red bikini": 1}}', 'ss_max_token_length': 'None', 'ss_ip_noise_gamma_random_strength': 'False', 'ss_face_crop_aug_range': 'None', 'ss_network_dropout': 'None', 'ss_datasets': '[{"is_dreambooth": true, "batch_size_per_device": 2, "num_train_images": 54, "num_reg_images": 0, "resolution": [1024, 1024], "enable_bucket": true, "min_bucket_reso": 512, "max_bucket_reso": 2048, "tag_frequency": {"image_dir": {"martha25": 54, "1girl": 54, "solo": 53, "long hair": 53, "breasts": 47, "looking at viewer": 47, "black hair": 54, "navel": 28, "cleavage": 35, "brown eyes": 52, "medium breasts": 31, "sitting": 12, "underwear": 12, "panties": 9, "indoors": 23, "dark skin": 27, "bra": 11, "white panties": 4, "dark-skinned female": 24, "lips": 47, "bed": 11, "arm support": 4, "on bed": 8, "underwear only": 9, "lingerie": 10, "lace trim": 5, "lace": 7, "freckles": 7, "white bra": 5, "realistic": 54, "lace panties": 6, "smile": 24, "barefoot": 12, "teeth": 3, "midriff": 7, "pants": 7, "necklace": 2, "grin": 10, "feet": 6, "crop top": 6, "pillow": 1, "window": 4, "toes": 4, "sunlight": 2, "pajamas": 1, "nose": 3, "bedroom": 6, "black panties": 4, "black bra": 5, "standing": 12, "full body": 9, "carpet": 1, "swimsuit": 23, "bikini": 23, "outdoors": 12, "day": 14, "water": 10, "black bikini": 12, "highleg": 13, "thigh gap": 1, "front-tie top": 3, "pool": 6, "highleg bikini": 11, "poolside": 4, "long sleeves": 4, "shorts": 5, "sweater": 4, "sleeves past wrists": 1, "short shorts": 5, "white shorts": 4, "white sweater": 4, "kitchen": 2, "photo background": 8, "sink": 1, "skirt": 2, "shirt": 4, "jewelry": 6, "white shirt": 2, "collared shirt": 2, "black skirt": 2, "chair": 3, "pencil skirt": 2, "office lady": 2, "computer": 1, "office": 1, "dress": 3, "bare shoulders": 3, "sleeveless": 1, "black footwear": 1, "black dress": 1, "high heels": 2, "hand on hip": 3, "sleeveless dress": 2, "short dress": 1, "table": 4, "plant": 3, "building": 5, "couch": 4, "city": 6, "potted plant": 1, "cityscape": 4, "skyscraper": 1, "open clothes": 2, "abs": 1, "bikini top only": 1, "white pants": 1, "grey pants": 1, "green bikini": 1, "sky": 1, "white dress": 2, "sunset": 3, "railing": 1, "1boy": 1, "solo focus": 1, "cup": 2, "crossed legs": 2, "cafe": 1, "tree": 3, "beach": 3, "blue bikini": 2, "palm tree": 1, "ocean": 4, "white bikini": 3, "watercraft": 2, "ship": 2, "boat": 2, "warship": 2, "ass": 1, "thong": 1, "shadow": 4, "walking": 6, "phone": 1, "sandals": 2, "cellphone": 1, "road": 3, "street": 2, "sand": 1, "waves": 1, "hood": 1, "hoodie": 1, "black pants": 2, "hood down": 1, "black hoodie": 1, "cropped hoodie": 1, "jacket": 1, "black shirt": 1, "denim": 3, "ground vehicle": 1, "motor vehicle": 1, "jeans": 2, "car": 1, "denim jacket": 1, "bangs": 1, "closed mouth": 2, "stomach": 1, "bare legs": 1, "on couch": 1, "cropped sweater": 2, "tank top": 2, "denim shorts": 1, "white tank top": 1, "upper body": 9, "earrings": 5, "outstretched arms": 5, "cowboy shot": 1, "highleg swimsuit": 1, "simple background": 3, "white background": 3, "profile": 2, "collarbone": 2, "nude": 2, "portrait": 1, "black eyes": 1, "thighs": 1, "blurry": 1, "book": 2, "lamp": 2, "navel piercing": 2, "large breasts": 2, "arms at sides": 1, "armpits": 1, "arm up": 1, "sports bra": 1, "gym": 1, "kneeling": 1, "faucet": 1, "holding": 1, "wooden floor": 1, "open book": 1, "reading": 1, "one eye closed": 1, "spread legs": 1, "toned": 1, "tan": 2, "highleg panties": 1, "lace bra": 1, "red bikini": 1}}, "bucket_info": {"buckets": {"0": {"resolution": [768, 1344], "count": 50}, "1": {"resolution": [896, 1152], "count": 3}, "2": {"resolution": [1024, 1024], "count": 1}}, "mean_img_ar_error": 0.003428109149200876}, "subsets": [{"img_count": 54, "num_repeats": 1, "color_aug": false, "flip_aug": false, "random_crop": false, "shuffle_caption": false, "keep_tokens": 1, "keep_tokens_separator": "", "secondary_separator": null, "enable_wildcard": false, "caption_prefix": null, "caption_suffix": null, "image_dir": "image_dir", "class_tokens": null, "is_reg": false}]}]', 'ss_num_train_images': '54', 'modelspec.date': '2026-03-07T11:12:41', 'ss_training_finished_at': '1772853161.4197257', 'modelspec.resolution': '1024x1024', 'ss_text_encoder_lr': '2e-05', 'ss_base_model_version': 'sdxl_base_v1-0', 'ss_huber_c': '0.1', 'modelspec.architecture': 'stable-diffusion-xl-v1-base/lora', 'sshs_legacy_hash': 'e097c2bf', 'ss_v2': 'False', 'ss_noise_offset_random_strength': 'False', 'modelspec.prediction_type': 'epsilon', 'ss_optimizer': 'bitsandbytes.optim.adamw.AdamW8bit', 'ss_ip_noise_gamma': 'None', 'modelspec.sai_model_spec': '1.0.0', 'ss_full_fp16': 'False', 'ss_noise_offset': '0.0', 'ss_scale_weight_norms': 'None', 'ss_network_module': 'networks.lora', 'ss_num_batches_per_epoch': '28', 'sshs_model_hash': '914d0e49cfa6c6ae1d66c92ae2d6e713dc6ec401589a3f75a99f5aab0fc6df97', 'ss_epoch': '57', 'ss_sd_model_name': 'EMS-14040-EMS.safetensors', 'ss_max_grad_norm': '1.0', 'ss_num_epochs': '57', 'ss_dataset_dirs': '{"image_dir": {"n_repeats": 1, "img_count": 54}}', 'ss_lowram': 'False', 'ss_network_dim': '64', 'ss_sd_model_hash': 'be9edd61', 'ss_output_name': 'lora', 'ss_seed': '303863893', 'ss_caption_dropout_every_n_epochs': '0', 'ss_gradient_checkpointing': 'True', 'ss_gradient_accumulation_steps': '1', 'ss_new_sd_model_hash': '31e35c80fc4829d14f90153f4c74cd59c90b779f6afe05a74cd6120b893f7e5b'}`

**Comment by Ankhesena** [comment]
had the same issue with unknown base model and the problem appears to be with line 261 in .py/utils/metadata_manager.py https://github.com/willmiao/ComfyUI-Lora-Manager/blob/ee84b30023f7e24559f5e8d5af86d6d31809656c/py/utils/metadata_manager.py#L261
uncommenting this line solved the issue for me

**Comment by willmiao** [comment]
> uncommenting this line solved the issue for me

That line was intended to resolve the base model using embedded training metadata (like `kohya_ss `data). However, I’ve found this method to be unreliable because that metadata is often missing or inaccurate, which leads to more user confusion than actual value.

That said, I agree that embedded training data is very useful for trigger word suggestions. This is already supported: if you click the pencil icon in the trigger words section to enter edit mode, you’ll see the training tags listed as suggestions, sorted by frequency.

<img width="872" height="482" alt="Image" src="https://github.com/user-attachments/assets/cafd4eec-672a-4f8b-8e70-4250e27a0b38" />

**Comment by jeremymeyers** [comment]
Is there any way to grab base model information this way?

**Comment by willmiao** [comment]
> Is there any way to grab base model information this way?

Unfortunately, grabbing base model information this way is very unreliable.

**Comment by willmiao** [comment]
Closing this for now. Feel free to reopen if further discussion is needed.
